sdk/python/kfp/_client.py

```diff
@@ -277,9 +289,32 @@ def set_user_namespace(self, namespace):
       namespace: kubernetes namespace the user has access to.
     """
     self._context_setting['namespace'] = namespace
+    if not os.path.exists(os.path.dirname(Client.LOCAL_KFP_CONTEXT)):
+        os.makedirs(os.path.dirname(Client.LOCAL_KFP_CONTEXT))
     with open(Client.LOCAL_KFP_CONTEXT, 'w') as f:
       json.dump(self._context_setting, f)
 
+  def get_kfp_healthz(self):
+    """Get healthz info of KFP deployment.
+
+    Returns:
+      response: json formatted response from the healtz endpoint.
+    """
+    healthz_api = 'http://' + self._existing_config.host + '/' + Client.HEALTH_PATH
+    count = 0
+    response = None
+    max_attempts = 60
+    while count < max_attempts and not response:
+      count += 1
+      try:
+        r = requests.get(healthz_api)
+        r.raise_for_status()
+        response = r.json()
+        return response
+      except ValueError:
+        print("Exception; retrying after 5 seconds", ValueError)
+        time.sleep(5)
```

Comment:
We should raise an timeout exception if the max attempts isn't enough.

Reply:
+1